third-party/lib/linenoise/linenoise.c

```diff
@@ -662,8 +662,10 @@ static int linenoiseEdit(int fdin, int fdout, char *buf, size_t buflen, const ch
         switch(c) {
         case 10:    /* '\n' enter */
         case 13:    /* '\r' enter */
-            history_len--;
-            sysfree(history[history_len]);
+	    if (history_len) {
```

Comment:
In some templates there were no heap in config, so this variable was NULL. Without MMU it produced no errors, but with MMU on it led to exception